### ISSUE TITLE
Vote on operator-linebreak rule

### ISSUE THREAD

**Issue by playpauseandstop** [ezhome/eslint-config-ezhome#9]
Right now in our code 2 different styles of `operator-linebreak` rule used,

`after`,

``` js
const myVar = yourVar.longMethodName() ?
  prettyValue :
  notSoPrettyValue;
if (some ||
    another) {
  ...
}
```

`before`,

``` js
const myVar = yourVar.longMethodName()
  ? prettyValue
  : notSorPrettuValue;
if (some
    || another) {
  ...
}
```

Let make a decision and figure out one and only style for `operator-linebreak` rule.

---

Please comment to this issue which style you prefer and I will add the final result to our `1.0.0` config.

**Comment by playpauseandstop** [comment]
`after`

**Comment by owais** [comment]
I find `before` more readable but don't have a strong preference. BTW, is there a rule to limit the size of the ternary operator? I personally feel it should be replaced with a simple if-else if it gets more complex than a one liner. 

(will be unavailable for an hour)

**Comment by liubko** [comment]
`before`

**Comment by dimkaufo** [comment]
`before`

**Comment by solex** [comment]
`after`

**Comment by playpauseandstop** [comment]
@owais,

There are 3 rules,
- `no-ternary` to fully disable ternary operator
- `no-unneeded-ternary` to avoid unnecessary ternary operator
- `no-nested-ternary` to avoid nested ternary operator

2 latest already enabled in ESLint config starting from [1.0.0-beta.2](https://github.com/ezhome/eslint-config-ezhome/blob/master/CHANGELOG.md#100-beta2-2016-06-22). Not sure that we need to enable first one and fully remove ternary operator from our frontend code.

**Comment by owais** [comment]
Sounds great. Completely removing it is a bit extreme :)

**Comment by vadimgoncharov** [comment]
ternary: `before`
if: `after`

---

I think for ternary `before` looks more readable, because we can recognize very fast where is `value_if_true` and where is `value_if_false` just by looking at first symbol:

```
const myVar = yourVar.longMethodName()
  ? prettyValue
  : notSorPrettuValue;
```

For `if`  both of options are hard to read, because of extra indentation that `if` added. However if we use `after` it would be more readable (if indent + condition indent VS if indent + condition indent + || indent)

```
    if (someCase ||
        anotherCase ||
        andTheLastThing
    ) {

    }

    if (someCase
        || anotherCase
        || andTheLastThing
    ) {

    }
```

For `if` it would better to not use multiline at all and make variable instead:

If we use `before`:

```
const wasClicked = (
  someCondition
  || anotherCondition
  || andTheLastThing
);

if (wasClicked) {

}
```

if we use `after`:

```
const wasClicked = (
  someCondition ||
  anotherCondition ||
  andTheLastThing
);

if (wasClicked) {

}
```

**Comment by anton-rodin** [comment]
`after`

**Comment by ehinter** [comment]
after

**Comment by playpauseandstop** [comment]
I will enable `after` by default and to change to `before` you'll need to put,

```
  "rules": {
    "operator-linebreak: [2, "before"]
  }
```

into `"eslintConfig"` section in `package.json`

**Comment by playpauseandstop** [comment]
Fixed in #10
